application-analytics-ui/src/main/resources/Analytics/Code/Macros/VelocityMacros.xml

```diff
@@ -61,5 +61,103 @@
   #end
   #set ($date = $startDate + ',' + $endDate)
 #end
+
+#macro(setParameterOnAction $parameter)
+&lt;div class="displayMostViewedPagesActions"  data-parameter=$escapetool.url($parameter)&gt;
+  &lt;a class="action" title="$escapetool.xml($services.localization.render(
+    'analytics.mostViwedMacros.action.rowEvolution'))" href="#"&gt;
+    &lt;span class="action-icon"&gt;$services.icon.renderHTML('eye')&lt;/span&gt;&lt;span class="action-label"&gt;
+    $escapetool.xml($services.localization.render('analytics.action.label.rowEvolution'))&lt;/span&gt;
+  &lt;/a&gt;
+&lt;/div&gt;
+#end
+
+#macro(modalRowEvolution $fields $requestPage)
+{{html clean=false}}
+&lt;div class="modal fade" id="chartModal" tabindex="-1" role="dialog" aria-labelledby="chartModalLabel" 
```

Comment:
`chartModal` is to generic and could crash with other id on the page. Be more specific, maybe `analyticsEvolutionChartModal`
But, if I have 2 mostViewedPages macros on the page, this `modalRowEvolution` will be added twice, and so will the id. Maybe you can do only with a specific class?

Will this macro be used for the evolution inside other matomo widgets? E.g. you told me about site search and how it uses the evolution action. Is this macro used for that as well, or there is a new macro created?

Reply:
> Will this macro be used for the evolution inside other matomo widgets? E.g. you told me about site search and how it uses the evolution action. Is this macro used for that as well, or there is a new macro created?

That was an oversight on my part. Right now I use the same macro, I should find a way to add a new modal for each macro and identify them